java-checks-test-sources/default/src/main/files/non-compiling/checks/NoTestInTestClassCheckSample.java

```diff
@@ -122,6 +122,16 @@ public class TestNGClassTest { // Noncompliant
   private void test1() { }
   public static void foo() {}
 }
+
+@org.testng.annotations.Test
+public class TestNGClassTestUseAnnotation {
```

Comment:
We can change the name of the class to `TestNGDoesNotRequirePublicTestMethods` or add a `compliant` comment explaining why this case is not an issue